src/main/java/com/laserfiche/repository/api/clients/impl/RepositoriesClientImpl.java

```diff
@@ -48,6 +51,52 @@ public RepositoryInfo[] getRepositoryList() {
                 throw ApiClientUtils.createApiException(httpResponse, problemDetails);
             }
         };
-        return ApiClientUtils.sendRequestWithRetry(httpClient, httpRequestHandler, baseUrl + "/v1/Repositories", "GET", null, null, null, null, null, null, new HashMap<String, String>(), false, parseResponse);
+        return ApiClientUtils.sendRequestWithRetry(httpClient, httpRequestHandler, baseUrl + "/v1/Repositories", "GET",
+                null, null, null, null, null, null, new HashMap<String, String>(), false, parseResponse);
+    }
+
+    /**
+     * Returns the repository resource list that current user has access to given the API server base URL. Only available in Laserfiche Self-Hosted.
+     *
+     * @param url API server base URL e.g., https://{APIServerName}/LFRepositoryAPI
+     * @return Get the respository resource list successfully.
+     */
+    public static RepositoryInfo[] getSelfHostedRepositoryList(String url) {
+        Map<String, String> headerKeyValuePairs = new HashMap<>();
+        headerKeyValuePairs.put("accept", "application/json");
+        HttpResponse<Object> httpResponse = null;
+        String responseJson;
+        if (url.endsWith("/")) {
+            url = url.substring(0, url.length() - 1);
+        }
+        String baseUrl = url + "/v1/Repositories";
+        ObjectMapper objectMapper = new TokenClientObjectMapper();
+        UnirestInstance httpClient = Unirest.spawnInstance();
```

Comment:
should dispose the httpClient afterwards. Could use a try-with-resource pattern on the httpClient
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
ok done